cfgmgr/coppmgr.cpp

```diff
@@ -450,68 +587,168 @@ void CoppMgr::doCoppTrapTask(Consumer &consumer)
             {
                 if (conf_present)
                 {
-                    SWSS_LOG_DEBUG("Deleting trap key %s", key.c_str());
-                    removeTrapIdsFromTrapGroup(m_coppTrapConfMap[key].trap_group,
-                            m_coppTrapConfMap[key].trap_ids);
-                    trap_ids.clear();
+                    removeTrap(key);
                     setCoppTrapStateOk(key);
-                    getTrapGroupTrapIds(m_coppTrapConfMap[key].trap_group, trap_ids);
-                    fvs.clear();
-                    FieldValueTuple fv(COPP_TRAP_ID_LIST_FIELD, trap_ids);
-                    fvs.push_back(fv);
-                    if (!checkTrapGroupPending(m_coppTrapConfMap[key].trap_group))
-                    {
-                        m_appCoppTable.set(m_coppTrapConfMap[key].trap_group, fvs);
-                        setCoppGroupStateOk(m_coppTrapConfMap[key].trap_group);
-                    }
-                    m_coppTrapConfMap.erase(key);
+
+                    m_coppDeletedTrapConfMap[key] = m_coppTrapConfMap[key];
```

Comment:
I still do not understand the reason for null_cfg.
If the null_cfg is present it has just one key value pair which is null, null. We don't support hdel. So If a user sets the config again the user has to set the complete configuration. 
I don't think we need this data structure.

Reply:
Done